### DISCUSSION TITLE
get a slim dockerfile not image

### DISCUSSION THREAD

**Discussion by ftalamo** [slimtoolkit/slim#309] [Q&A]
HI, I am doing a project where I generate a dockerfile that I run in argo-cd to build the image in a cloud kubernetes cluster. I have tried docker-slim locally and it has worked great, however I don't know how to upload just a dockerfile with my slim image instructions to be built in cloud environments.

Any help will be welcome.

**Comment by kcq** [comment]
@ftalamo what do you mean by `my slim image instructions`? Do you mean you want to run docker-slim with your Dockerfile in Argo-CD?

**Reply by ftalamo** [reply]
I need to generate my docker-image in Argo-CD I just upload the docker file. So I don't know how to slim the image on Argo.

**Reply by HJX-zhanS** [reply]
@ftalamo Hello, have you solved the problem? I have a similar demand at present, and I need to check the `Dockerfile` after slim.

**Reply by kcq** [reply]
you can use the `--image-build-engine` flag setting the value to `docker` and that should produce a Dockerfile for the minified image (in the artifacts directory).

**Reply by HJX-zhanS** [reply]
Thanks for your reply. However, when I use `docker-slim build --image-build-engine docker --http-probe=false <image_id>` to build the image, I get an error `API /build requires version 1.32, requested version 1.25 is insufficient.`

docker-slim version: 1.40.11
docker version: 25.0.3

**Reply by kcq** [reply]
@HJX-zhanS try [the latest version](https://github.com/mintoolkit/mint/releases/tag/1.41.1)

**Reply by HJX-zhanS** [reply]
@kcq I have tried the latest version, but this issue still exists.

**Reply by SunBha** [reply]
@kcq I also tried the command --image-build-engine docker , and same error is coming 
API /build requires version 1.32, requested version 1.25 is insufficient

my versions are
docker engine ver: 24.0.7
slim version : 1.40.11

**Reply by kcq** [reply]
@SunBha the `--image-build-engine` flag is for the output image build step. The original poster tried to use the app to build the input image too, so unless you are trying to do the same thing you can ignore the `--image-build-engine` flag. Please use [this app version](https://github.com/mintoolkit/mint/releases/tag/1.41.1) to make sure the right Docker API version is used. Either way, you can also specify the Docker API version to use with the `--crt-api-version` global flag when you run the app or you can also try setting the `DOCKER_API_VERSION` environment variable.

**Reply by SunBha** [reply]
@kcq Thank you for the information. I read your other discussions and figured out the way by using this command:

slim --crt-api-version 1.32 build  --image-build-engine docker ..............

**Reply by kcq** [reply]
great!
